### PR TITLE
FEAT: Support Markdown Rendering

### PR DESCRIPTION
**Pull request by tahierhussain**
## What

Introduced a `CustomMarkdown` component for selective markdown rendering, processing specific markdown syntax such as bold, newline, links, and codeblocks while displaying unsupported syntax as plain text. This component is used in three places: popup notification messages, logs component, and RJSF form's helper text.

## Why

To make messages more interactive and user-friendly. Previously, the displayed messages were less engaging and harder to read.

## How

The `CustomMarkdown` component is designed to selectively parse and render supported markdown syntax while leaving unsupported syntax as-is. It uses a parsing logic that identifies markdown elements (like **bold**, [link](url), and ```code```) and converts them into appropriate React elements for rendering. This ensures consistent and user-friendly message formatting across all integrated components.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

Yes, the `CustomMarkdown` component is used in multiple places, which increases the chances of it potentially breaking existing features.

## Database Migrations

NA

## Env Config

NA

## Relevant Docs

NA

## Related Issues or PRs

https://github.com/Zipstack/unstract-sdk/pull/133

## Dependencies Versions

NA

## Notes on Testing

NA

### CustomMarkdown component used inside the Notifications and Logs component:
Screenshot from 2024-12-10 16-50-43

### CustomMarkdown component used inside the RjsfFormLayout component:
Screenshot from 2024-12-10 16-51-09

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by vishnuszipstack — APPROVED** [review]
Left some minor comments. everything else looks okay.,

**Review by chandrasekharan-zipstack — APPROVED** [review]
Tested the FE changes - haven't review the FE code though. Would suggest to get another approval for the same

File: frontend/src/components/pipelines-or-deployments/log-modal/LogsModal.jsx [hunk 1]
```diff
@@ -40,7 +41,7 @@ const LogsModal = ({
       .then((res) => {
         const logDetails = res.data.results.map((item) => ({
           id: item.id,
```
Comment (jagadeeswaran-zipstack) [inline frontend/src/components/pipelines-or-deployments/log-modal/LogsModal.jsx:43]: Can we add optional chaining wherever required.

**Review by jagadeeswaran-zipstack — CHANGES_REQUESTED** [review]

File: frontend/src/components/rjsf-custom-widgets/checkbox-widget/CheckboxWidget.jsx [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
     onChange(event.target.checked);
   };
 
   return (
-    <Space className="checkbox-widget-main">
+    <Space direction="vertical" className="checkbox-widget-main" size={0}>
       <Checkbox id={id} checked={value} onChange={handleCheckboxChange}>
-        <Typography>{label}</Typography>
+        <Typography>
+          {required && <span className="form-item-required">* </span>}
```
[the remark this answers is not in the document]
Comment (tahierhussain) [reply frontend/src/components/rjsf-custom-widgets/checkbox-widget/CheckboxWidget.jsx:15]: Yes, @vishnuszipstack. It is intentional.

File: frontend/src/components/pipelines-or-deployments/log-modal/LogsModal.jsx (hunk 1 above)
Comment (tahierhussain) [reply frontend/src/components/pipelines-or-deployments/log-modal/LogsModal.jsx]: @jagadeeswaran-zipstack Thanks for the catch. Done. I have added optional chaining in places where it was necessary.

**Review by jagadeeswaran-zipstack — APPROVED** [review]

File: frontend/src/components/helpers/custom-markdown/CustomMarkdown.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        nextIndex = Math.min(...nextSpecialIndices);
+      }
+
+      const textSegment = input.substring(index, nextIndex);
+      elements.push(textSegment);
+      index = nextIndex;
+    }
+
+    return elements;
+  }, [text, renderNewLines, textType, className]);
```
[the remark this answers is not in the document]
Comment (chandrasekharan-zipstack) [reply frontend/src/components/helpers/custom-markdown/CustomMarkdown.jsx]: Resolving since this is outdated now

**Review by vishnuszipstack — APPROVED** [review]

**Pull request merged**